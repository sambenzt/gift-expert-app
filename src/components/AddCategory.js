import React, { useState } from 'react'
import PropTypes from 'prop-types';    

export const AddCategory = ({ setCategories }) => {

    const [value, setValue] = useState('');
    
    const handleInputChange = (e) => {
        setValue(e.target.value);
    }

    const handleForm = (e) => {
        e.preventDefault();

        if(value.trim().length > 2) {

            setCategories(categories => [value, ...categories]);
            setValue('');
        }

    }

    return (
        <form onSubmit={ handleForm }>
            <input 
                onChange={ handleInputChange }
                type="text" 
                value={ value }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
