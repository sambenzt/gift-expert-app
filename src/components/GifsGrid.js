

import React from 'react'
import { GifsGridItem } from './GifsGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifsGrid = ({ category }) => {
  
    const { data:gifs, loading } = useFetchGifs(category);
  
    return (
      <>
        <h3>{ category }</h3>

        <p className='animate__animated animate__flash'>{ loading ? 'cargando...' : '' }</p>

        <div className='card-grid animate__animated animate__delay-4s'>
            {
              gifs.map(gif => (

                <GifsGridItem 
                  key={ gif.id } 
                  {...gif}
                />

              ))
            }
        </div>
      </>
    )
}
