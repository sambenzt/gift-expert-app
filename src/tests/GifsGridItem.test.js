import React from 'react';
import { shallow } from "enzyme";
import { GifsGridItem } from '../components/GifsGridItem';


describe('GifsGrid', () => { 

    test('se renderiza', () => { 

        const wrapper = shallow(<GifsGridItem />);

        expect( wrapper ).toMatchSnapshot();

    });

});