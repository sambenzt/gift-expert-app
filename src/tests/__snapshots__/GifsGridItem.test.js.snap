// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifsGrid se renderiza 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
